### PR TITLE
add plunderswap dex volume

### PR DESCRIPTION
**Pull request by darthgus**
add plunderswap dex volume

### CODE REVIEW

**Comment by llamatester** [comment]
The plunderswap adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts dexs plunderswap
    
    🦙 Running PLUNDERSWAP adapter 🦙
    ---------------------------------------------------
    Start Date:	Tue, 10 Dec 2024 00:00:00 GMT
    End Date:	Wed, 11 Dec 2024 00:00:00 GMT
    ---------------------------------------------------
    
    ZILLIQA 👇
    Backfill start time: 10/12/2024
    Daily volume: 8.51 k
    └─ Methodology: Volume of trades on Plunderswap at the start of the day (12:00:00 UTC) for the previous day
    End timestamp: 1733788800 (2024-12-10T00:00:00.000Z)

**Comment by darthgus** [comment]
@dtmkeng i apologize greatly for the previous PR issue.  wont happen again.

File: dexs/plunderswap/index.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+const fetch = async (timestamp: number, _: ChainBlocks, { createBalances, startOfDay, }: FetchOptions) => {
+  const dailyVolume = createBalances()
+  const dayTimestamp = getUniqStartOfTodayTimestamp(new Date(timestamp * 1000))
+  const historicalVolume: IVolumeall[] = (await fetchURL(historicalVolumeEndpoint));
+  
+  // Find the entry closest to the start of day
+  const closestEntry = historicalVolume
+    .reduce((closest, current) => {
```
Comment (dtmkeng) [inline dexs/plunderswap/index.ts]: i think you can filter time more than start time and less than end time instead before reduce

**Review by dtmkeng — CHANGES_REQUESTED** [review]

**Comment by llamatester** [comment]
The plunderswap adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts dexs plunderswap
    
    🦙 Running PLUNDERSWAP adapter 🦙
    ---------------------------------------------------
    Start Date:	Tue, 10 Dec 2024 00:00:00 GMT
    End Date:	Wed, 11 Dec 2024 00:00:00 GMT
    ---------------------------------------------------
    
    ZILLIQA 👇
    Backfill start time: 10/12/2024
    Daily volume: 8.87 k
    └─ Methodology: Volume of trades on Plunderswap at the start of the day (12:00:00 UTC) for the previous day
    End timestamp: 1733788800 (2024-12-10T00:00:00.000Z)

File: dexs/plunderswap/index.ts (hunk 1 above)
Comment (darthgus) [reply dexs/plunderswap/index.ts]: Ok good point, think I have implemented that correctly.

**Comment by llamatester** [comment]
The plunderswap adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts dexs plunderswap
    
    🦙 Running PLUNDERSWAP adapter 🦙
    ---------------------------------------------------
    Start Date:	Tue, 10 Dec 2024 00:00:00 GMT
    End Date:	Wed, 11 Dec 2024 00:00:00 GMT
    ---------------------------------------------------
    
    ZILLIQA 👇
    Backfill start time: 10/12/2024
    Daily volume: 8.98 k
    └─ Methodology: Volume of trades on Plunderswap at the start of the day (12:00:00 UTC) for the previous day
    End timestamp: 1733788800 (2024-12-10T00:00:00.000Z)

**Comment by llamabutler** [comment]
The plunderswap adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts dexs plunderswap
    
    🦙 Running PLUNDERSWAP adapter 🦙
    ---------------------------------------------------
    Start Date:	Thu, 12 Dec 2024 00:00:00 GMT
    End Date:	Fri, 13 Dec 2024 00:00:00 GMT
    ---------------------------------------------------
    
    ZILLIQA 👇
    Backfill start time: 10/12/2024
    Daily volume: 5.38 k
    └─ Methodology: Volume of trades on Plunderswap at the start of the day (00:00:00 UTC) for the previous day
    End timestamp: 1733961600 (2024-12-12T00:00:00.000Z)

**Comment by darthgus** [comment]
Having a look over my working out, i seen it was still inefficient, so have re-written it much more simpler now.    will give the exact value based on 00:00:00 entry. @dtmkeng

**Pull request merged**